server/src/main/java/org/candlepin/async/JobManager.java

```diff
@@ -982,19 +982,20 @@ public AsyncJobStatus findJob(String jobId) {
      *  an AsyncJobStatus instance representing the queued job's status, or the status of the
      *  existing job if it already exists
      */
-    public synchronized AsyncJobStatus queueJob(JobConfig config) throws JobException {
-
-        ManagerState state = this.getManagerState();
-        if (state != ManagerState.RUNNING) {
-            // Check if we're paused. If so, and if the "queue while paused" config is not set,
-            // throw our usual ISE
-            if (state != ManagerState.SUSPENDED ||
-                !this.configuration.getBoolean(ConfigProperties.ASYNC_JOBS_QUEUE_WHILE_SUSPENDED)) {
-
-                String msg = String.format("Jobs cannot be queued while the manager is in the %s state",
-                    state);
-
-                throw new IllegalStateException(msg);
+    public AsyncJobStatus queueJob(JobConfig config) throws JobException {
+        synchronized (this) {
```

Comment:
@nikosmoum 
If the same JobManager instance shared with multiple Artemis threads and the intention is to run `queueJob` and `executeJob` methods in parallel then `synchronized (this)` will not help here as both blocks are synchronized over `this`. 

I would recommend using 2 different locks in these methods. You can try using 2 different `ReentrantLock` here.  

Also, I did not get how the synchronized block will help to run jobs in parallel. 

Reply:
@Ceiu The `synchronized (this)` block does not span the whole queueJob and executeJob methods. That would be pointless because it would be the same as leaving the synchronized keyword on the method signature itself. It only synchronizes the little bit of code that checks the manager state. when they get to the `result = job.execute(status);` bit, there is no synchronization. But now I realize that the `getManagerState()` itself is synchronized, so there is no point in even doing that.

@abhiskum I am aware that they will be synchronized over the same object, and it does make sense in this case. Instead of synchronizing the whole method, only this small bit that checks the manager state is now synchronized (and we want it to be synchronized over the same object, because we're checking the state of the JobManager, which is a Singleton), not the whole of execute and queue methods. But as I said before, I realize this is pointless because that method we're calling is already synchronized, so I'll remove this.

Reply:
Thank you, @nikosmoum for details.